### PR TITLE
[Perf] Upgrade perf pipeline to Java 17

### PR DESCRIPTION
**Pull request by mikeharder**
# Validation Builds
- storage-blob: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2128947&view=results
- storage-file-datalake: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2129129&view=results
- storage-file-share: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2129130&view=results

### CODE REVIEW

File: eng/pipelines/templates/jobs/perf.yml [hunk 1]
```diff
@@ -33,10 +33,10 @@ extends:
     InstallLanguageSteps:
     - task: JavaToolInstaller@0
       inputs:
-        versionSpec: '8'
+        versionSpec: '17'
```
Comment (alzimmermsft) [inline eng/pipelines/templates/jobs/perf.yml:36]: Could we make this something that is configurable in the pipeline with a default of Java 17?

Comment (mikeharder) [reply eng/pipelines/templates/jobs/perf.yml:36]: Good suggestion, I will do this for all languages.  I also need to make some changes to the perf automation.  I was originally planning to have the perf automation app control the language version (which works for .NET), but for other languages we usually configure the language version in the pipeline YML.  I will change the perf automation app to just **report** the language version used, not try to **select** the language version used.

**Review by g2vinay — APPROVED** [review]

**Comment by JonathanGiles** [comment]
@mikeharder Can this be merged?

**Comment by mikeharder** [comment]
@alzimmermsft, @g2vinay: Do you want to move Java to 17 now?  I'm planning to move the other languages to latest eventually, but they're not ready yet.  So we can either move Java now or wait for the other languages.

**Comment by alzimmermsft** [comment]
> @alzimmermsft, @g2vinay: Do you want to move Java to 17 now? I'm planning to move the other languages to latest eventually, but they're not ready yet. So we can either move Java now or wait for the other languages.

I have no concern on choosing either direction to go, be it pinning to Java 17 and being ahead of the other languages or waiting for all languages to support multiple runtimes at once.

**Comment by mikeharder** [comment]
@alzimmermsft: OK, I'll do quick runs of blob perf with both versions of Java to check for changes, then merge this.

**Comment by mikeharder** [comment]
Converting back to draft, since we are getting slower results for Java 17 in some scenarios.  Further testing is required to determine if these are real regressions or just noise.

**Pull request merged**